### PR TITLE
Added temp dir for test files

### PR DESCRIPTION
**Pull request by jessemapel**
## Description

This creates a temporary directory to house temp files generated during tests. The temporary directory is then cleaned up once the test exits. I also changed the test cube files so that they are generated with the proper suffix.

## Related Issue

No related issue, this is part of the GTest conversions.

## Motivation and Context

The current implementation sometimes leaves .cub files dangling. This will also be important when testing applications like jigsaw that generate a range of files in the same directory.

## How Has This Been Tested?

Existing GTests have all been tested and still work. Makefile and old unit tests are un-affected.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jessemapel** [comment]
Once this is merged #3693, #3682, and #3688 can be updated. Or if they are merged before this I can update the code.

File: isis/tests/Fixtures.cpp [hunk 1]
```diff
@@ -2,14 +2,26 @@
...
 
+  void TempTestingFiles::SetUp() {
+    ASSERT_TRUE(tempDir.isValid());
+  }
+
+
   void DefaultCube::SetUp() {
+    TempTestingFiles::SetUp();
+
+    tempFile.setFileTemplate(tempDir.path() + "/XXXXXX.cub");
```
Comment (acpaquette) [inline isis/tests/Fixtures.cpp]: Are the templates still necessary if the temp directory as a whole (files included) will be cleaned up or am I missing something?

**Review by acpaquette — COMMENTED** [review]
Other than the one comment this looks good!

File: isis/tests/Fixtures.cpp (hunk 1 above)
Comment (jessemapel) [reply isis/tests/Fixtures.cpp]: The template is required to put the file in the temp directory and by adding `.cub` we ensure that we only make one file.

**Review by scsides — DISMISSED** [review]

**Review by AustinSanders — DISMISSED** [review]

**Review by kaitlyndlee — DISMISSED** [review]
Looks good!

File: isis/tests/Fixtures.cpp (hunk 1 above)
Comment (Kelvinrr) [reply isis/tests/Fixtures.cpp]: I don't see where in the docs it says the files under the temp dir need to be tempfile objs. It seems like any file in the directory gets cleaned up regardless of how the file got there. Why not create the temp dir, save the path (or tempdir object) as a member of the fixure, and make the underlying file names match 1-1 with what is in the test/data/<fixture> area?

Comment (jessemapel) [reply isis/tests/Fixtures.cpp]: The files in the tempdir don't need to be tempfiles also. I'll change these to just regular files.

File: isis/tests/cnetcheckFunctionalTests.cpp [hunk 2]
```diff
@@ -76,27 +72,24 @@ TEST_F(ThreeImageNetwork, FunctionalTestCnetcheckNoPoints) {
   // remove all controlpoints from test network
   QList<ControlPoint*> cpoints = network->GetPoints();
 
-  for (size_t i = 0; i<cpoints.size(); i++) {
+  for (int i = 0; i<cpoints.size(); i++) {
```
Comment (Kelvinrr) [inline isis/tests/cnetcheckFunctionalTests.cpp:75]: Why int? In this case, it will probably not matter but seems like a weird change.

**Review by Kelvinrr — DISMISSED** [review]
Only a benign comment in-line.

File: isis/tests/cnetcheckFunctionalTests.cpp (hunk 2 above)
Comment (jessemapel) [reply isis/tests/cnetcheckFunctionalTests.cpp:75]: It raised a warning because QT containers return ints for size instead of unsigned ints.

**Comment by Kelvinrr** [comment]
My vote right now would be to get @acpaquette's PRs merged and then update this if no one has an issue with it. #3682 only needs one quick change to be mergeable. #3682 I trust will be mergeable by the end of the day.

**Comment by Kelvinrr** [comment]
Alright, change of plans. PRs are taking too long to merge and I'd rather not make it Jesse's problem. I am going to merge this as it is also fixing a bug in the original DefaultCube Fixture and everyone else will update their PR. I'll update part-timer's PRs.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**